src/server.rs

```diff
@@ -813,12 +804,65 @@ impl WalletService for Arc<crate::wallet::Wallet> {
 
     async fn create_deposit_transaction(
         &self,
-        _request: tonic::Request<CreateDepositTransactionRequest>,
+        request: tonic::Request<CreateDepositTransactionRequest>,
     ) -> std::result::Result<tonic::Response<CreateDepositTransactionResponse>, tonic::Status> {
-        Err(tonic::Status::new(
-            tonic::Code::Unimplemented,
-            "not implemented",
-        ))
+        let CreateDepositTransactionRequest {
+            sidechain_id,
+            address,
+            value_sats,
+            fee_sats,
+        } = request.into_inner();
+
+        let value_sats = Amount::from_sat(value_sats);
+        let fee_sats = match fee_sats {
+            0 => None,
+            fee => Some(Amount::from_sat(fee)),
+        };
+
+        let sidechain_id: SidechainNumber = {
+            <u8 as TryFrom<_>>::try_from(sidechain_id)
+                .map_err(|_| {
+                    invalid_field_value::<CreateDepositTransactionRequest>(
+                        "sidechain_id",
+                        &sidechain_id.to_string(),
+                    )
+                })?
+                .into()
+        };
+
+        if value_sats.to_sat() == 0 {
+            return Err(invalid_field_value::<CreateDepositTransactionRequest>(
+                "value_sats",
+                &value_sats.to_string(),
+            ));
+        }
+
+        if address.is_empty() {
+            return Err(invalid_field_value::<CreateDepositTransactionRequest>(
+                "address", &address,
+            ));
+        }
+
+        if !self
+            .is_sidechain_active(sidechain_id)
+            .map_err(|err| err.into_status())?
+        {
+            return Err(tonic::Status::new(
+                tonic::Code::FailedPrecondition,
+                format!("sidechain {sidechain_id} is not active"),
+            ));
+        }
+
+        let txid = self
+            .create_deposit(sidechain_id, address, value_sats, fee_sats)
+            .await
+            .map_err(|err| err.into_status())?;
+
+        let txid = convert::bdk_txid_to_bitcoin_txid(txid);
```

Comment:
Probably best to return `bitcoin::Txid` from `self.create_deposit`. `bdk` types should be limited to interactions with `bdk` if possible